common/common.cpp

```diff
@@ -2076,6 +2077,17 @@ static bool llama_download_file(const std::string & url, const std::string & pat
             }
         }
 
+        // Create parent directories if not exist
+        const std::vector<std::string> path_parts = string_split(path_temporary, DIRECTORY_SEPARATOR);
+        std::string parent_dir = "";
+        struct stat st;
+        for (unsigned i = 0; i < path_parts.size() - 1; i++) {
+            parent_dir += path_parts[i] + DIRECTORY_SEPARATOR;
+            if (stat(parent_dir.c_str(), &st) != 0) {
+                mkdir(parent_dir.c_str(), S_IRWXU);
```

Comment:
Probably we should check the return value of `mkdir`

Reply:
Done.

Reply:
As @ggerganov , `create_directory_with_parents` will create so maybe here you don't need to do that anymore